apps/src/p5lab/spritelab/SpritelabInput.jsx

```diff
@@ -64,30 +74,84 @@ class SpritelabInput extends React.Component {
     collapsed: false
   };
 
-  userInputSubmit() {
+  userInputSubmit(value) {
     const variableName =
       this.props.inputList[0] && this.props.inputList[0].variableName;
     if (!variableName) {
       return;
     }
     this.props.onPromptAnswer();
-    coreLibrary.onPromptAnswer(variableName, this.state.userInput);
+    coreLibrary.onPromptAnswer(variableName, value);
+  }
+
+  onTextSubmit() {
+    this.userInputSubmit(this.state.userInput);
     this.setState({userInput: ''});
   }
 
+  onMultipleChoiceSubmit(e) {
+    this.userInputSubmit(e.target.value);
+  }
+
   toggleCollapsed = () =>
     this.setState({
       collapsed: !this.state.collapsed
     });
 
   render() {
+    const inputInfo = this.props.inputList[0];
```

Comment:
can we document the expected shape of this prop somewhere (possibly above where we set `propTypes`)? ideally, the shape could be dictated by `PropTypes`, but i'm not sure if that's possible in this case because each item in `this.props.inputList` may have a different shape